src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,265 @@
+        nullLoginAndPasswordUser.setLogin(null);
+        nullLoginAndPasswordUser.setPassword(null);
+        nullLoginAndPasswordUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class, () -> service.register(nullLoginAndPasswordUser));
+    }
+
+    @Test
+    public void add_emptyLoginUser_notOk() {
+        User emptyLoginUser = new User();
+        emptyLoginUser.setLogin(EMPTY_LINE);
+        emptyLoginUser.setPassword(VALID_PASSWORD);
+        emptyLoginUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class, () -> service.register(emptyLoginUser));
+    }
+
+    @Test
+    public void add_emptyPasswordUser_notOk() {
+        User emptyPasswordUser = new User();
+        emptyPasswordUser.setLogin(VALID_LOGIN);
+        emptyPasswordUser.setPassword(EMPTY_LINE);
+        emptyPasswordUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class, () -> service.register(emptyPasswordUser));
+    }
+
+    @Test
+    public void add_emptyLoginAndPasswordUser_notOk() {
+        User emptyLoginAndPasswordUser = new User();
+        emptyLoginAndPasswordUser.setLogin(EMPTY_LINE);
+        emptyLoginAndPasswordUser.setPassword(EMPTY_LINE);
+        emptyLoginAndPasswordUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class,
+                () -> service.register(emptyLoginAndPasswordUser));
+    }
+
+    @Test
+    public void add_oneSymbolLoginUser_notOk() {
+        User oneSymbolLoginUser = new User();
+        oneSymbolLoginUser.setLogin(ONE_SYMBOL_LINE);
+        oneSymbolLoginUser.setPassword(VALID_PASSWORD);
+        oneSymbolLoginUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class, () -> service.register(oneSymbolLoginUser));
+    }
+
+    @Test
+    public void add_threeSymbolsLoginUser_notOk() {
+        User threeSymbolLoginUser = new User();
+        threeSymbolLoginUser.setLogin(THREE_SYMBOLS_LINE);
+        threeSymbolLoginUser.setPassword(VALID_PASSWORD);
+        threeSymbolLoginUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class, () -> service.register(threeSymbolLoginUser));
+    }
+
+    @Test
+    public void add_fiveSymbolsLoginUser_notOk() {
+        User fiveSymbolLoginUser = new User();
+        fiveSymbolLoginUser.setLogin(FIVE_SYMBOLS_LINE);
+        fiveSymbolLoginUser.setPassword(VALID_PASSWORD);
+        fiveSymbolLoginUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class, () -> service.register(fiveSymbolLoginUser));
+    }
+
+    @Test
+    public void add_oneSymbolPasswordUser_notOk() {
+        User oneSymbolPasswordUser = new User();
+        oneSymbolPasswordUser.setLogin(VALID_LOGIN);
+        oneSymbolPasswordUser.setPassword(ONE_SYMBOL_LINE);
+        oneSymbolPasswordUser.setAge(VALID_AGE);
+        assertThrows(RegistrationException.class, () -> service.register(oneSymbolPasswordUser));
+    }
+
+    @Test
+    public void add_threeSymbolsPasswordUser_notOk() {
```

Comment:
Do you really need this check for 1, 3 and 5 symbols password/login ect? Why you have no checks for 100, -1000000000000, 3423423423, or 321?

Reply:
In your tests very important to check all possible edge cases. For example, we have min password length 6 characters. In this case would be good to test: empty password (0-length - not-ok), ‘abc’ (3-length - not-ok), ‘abcdf’ (5-length - not-ok edge case), ‘abcdef’ (6-length - ok edge case), ‘abcdefgh’ (8-length - ok). Few not-ok checks you can have in one not-ok test method. Same for ok-test. Testing only ‘abc’ (3-length - not-ok) and ‘abcdefgh’ (8-length - ok) is not enough because we do not check how the program behave with edge cases.

Reply:
Or did you mean, that I should add more?